### PR TITLE
Feature/localization number filters

### PR DESCRIPTION
**Pull request by jaynarayan89**
Open for review

### CODE REVIEW

File: system/View/NumberFilters.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+			'int64'    => \NumberFormatter::TYPE_INT64,
+			'double'   => \NumberFormatter::TYPE_DOUBLE,
+			'currency' => \NumberFormatter::TYPE_CURRENCY,
+		];
+
+			$formatter = self::getNumberFormatter($locale, $style);
+		if (! isset($typeValues[$type]))
+		{
+			throw new \Exception('syntax error');
+			//todo: raise mor appropriate /specific exception
```
Comment (lonnieezell) [inline system/View/NumberFilters.php]: Were you going to add more errors here?

File: system/View/NumberFilters.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	 * @param $value
+	 *
+	 * @return string
+	 */
+	private static function getNumberFormatter($locale, $style):\NumberFormatter
+	{
+		static $formatter, $currentStyle;
+
+		$locale = $locale ?? \CodeIgniter\Config\Services::request()->getLocale();
+log_message('info',\CodeIgniter\Config\Services::request()->getLocale());
```
Comment (lonnieezell) [inline system/View/NumberFilters.php]: This looks like debug code. Please remove.

File: system/View/NumberFilters.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	 *
+	 * @param $value
+	 *
+	 * @return string
+	 */
+	public static function localizedCurrency($number, $currency = null, $locale = null)
+	{
+		$locale = $locale ?? \CodeIgniter\Config\Services::request()->getLocale();
+
+			$formatter = self::getNumberFormatter($locale, 'currency');
```
Comment (lonnieezell) [inline system/View/NumberFilters.php]: Indentation is off.

File: system/View/NumberFilters.php [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+			'percent'    => \NumberFormatter::PERCENT,
+			'scientific' => \NumberFormatter::SCIENTIFIC,
+			'spellout'   => \NumberFormatter::SPELLOUT,
+			'ordinal'    => \NumberFormatter::ORDINAL,
+			'duration'   => \NumberFormatter::DURATION,
+		];
+		if (! isset($styleValues[$style]))
+		{
+			throw new \Exception('syntax error');
+			//todo: raise more appropriate /specific exception
```
Comment (lonnieezell) [inline system/View/NumberFilters.php]: More specific exceptions needed?

File: system/View/NumberFilters.php [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	//--------------------------------------------------------------------
+
+	/**
+	 * describe metho
+	 */
+	private static function raiseExceptionIfIntlExtensionNotAvailable()
+	{
+		if (! class_exists('IntlDateFormatter'))
+		{
+			throw new \RuntimeException('The intl extension is needed to use intl-based filters.');
```
Comment (lonnieezell) [inline system/View/NumberFilters.php]: This should be localized through the `lang()` function.

**Comment by jaynarayan89** [comment]
@lonnieezell I was waiting for your reply. Actually after writing this code I notice the format_number function of number helper which I think does the same thing (correct me if I wrong). So I wait before working further.

**Comment by lonnieezell** [comment]
Doh! You're right that `format_number` could be used. Silly me. However, it couldn't be used directly and would need to be called from a filter, since the parameter list is different. And it can throw an exception that you might want to capture, log, and return the original number?

**Comment by jaynarayan89** [comment]
@lonnieezell  I would like to suggest that we should 

1) create a  class  i18n\number and  use that class in number helper and localized filter. This way The NumberFIlters class  I had created using twig extension can be used in both files .

2)Adjust the number_filter helper function  to use  i18n\number class.

**Comment by lonnieezell** [comment]
@jaynarayan89 I'm fine with that.

**Comment by lonnieezell** [comment]
@jaynarayan89 Were you going to make these changes you suggested?

**Comment by jaynarayan89** [comment]
@lonnieezell  yes I am working on it. Please give me two or three days.

**Comment by lonnieezell** [comment]
No worries - just checking in it this had been a few weeks.

**Comment by lonnieezell** [comment]
This was never updated. I'm working on a commit for these.

**Pull request closed without merging**